azure-iot-device/tests/common/pipeline/test_pipeline_stages_mqtt.py

```diff
@@ -159,168 +222,219 @@ def test_sets_transport_attribute_on_root(self, stage, transport, op_set_connect
         stage.run_op(op_set_connection_args)
         assert stage.previous.transport == transport.return_value
 
-    @pytest.mark.it("Completes with success if no exception")
+    @pytest.mark.it("Sets the pending connection op tracker to None")
+    def test_pending_conn_op(self, stage, transport, op_set_connection_args):
+        stage.run_op(op_set_connection_args)
+        assert stage._pending_connection_op is None
```

Comment:
grumble, grumble, testing a private field, blah blah blah.

Reply:
Yeah, but how else are you going to test this? The entire mechanics of pending operations is handled in private.

At least we're looking at the private field from within the relevant unit. I'd be much more worried if we had to access private fields like this while testing a different unit.

---
In reply to: [311656186](https://github.com/Azure/azure-iot-sdk-python-preview/pull/159#discussion_r311656186) [](ancestors = 311656186)